backend/Tournament/views.py

```diff
@@ -0,0 +1,131 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from django.contrib.auth import authenticate
+from django.utils.decorators import method_decorator
+
+from .models import Tournament, \
+						TournamentPlayer
+from .serializers import TournamentSerializer, \
+	 						TournamentCreationSerializer, \
+							TournamentPlayerSerializer, \
+							TournamentInProgressSerializer
+from .managers import TournamentSetupManager, \
+						TournamentInProgressManager
+from .exceptions import TournamentSetupException, \
+							TournamentInProgressException
+from .tournamentState import TournamentState
+from User.models import User
+from shared_utilities.decorators import must_be_authenticated, \
+											must_not_be_username, \
+											check_that_valid_tournament_request
+
+import sys
+
+# Create your views here.
+class TournamentListView(APIView):
+	@method_decorator(must_be_authenticated)
+	def post(self, request):
+		body_data = request.data.copy()
+		body_data['host_user'] = request.user.id
+
+		tournament_creation_serializer = TournamentCreationSerializer(data=body_data)
+		if not tournament_creation_serializer.is_valid():
+			return Response({'error': 'invalid request for tournament creation'}, status=status.HTTP_400_BAD_REQUEST)
+		
+		try:
+			tournament = TournamentSetupManager.create_tournament_and_tournament_player_for_host(tournament_creation_serializer.validated_data)
+			host_tournament_player_serializer = TournamentPlayerSerializer(tournament.players.all().first())
+			return Response({
+				'tournament_id': tournament.id,
+				'tournament_player': host_tournament_player_serializer.data
+			}, status=status.HTTP_201_CREATED) 
+		except Exception as e:
+			return Response({'error': str(e)},status=status.HTTP_403_FORBIDDEN)
```

Comment:
Forbidden is not a solve-all exception. What kind of exceptions can happen within this try block? catch those and depending on the exceptions return a proper status code.

Reply:
the appropriate code would be 500, but we can't use that because of the eval. I'm using 403 in its stead, but if there is something more appropriate than that I'll use it

Reply:
500 Is not appropriate way to ever answer from the Rest API - It means 'Oops something we didnt handle or think about is returned', as discussed multiple times before.

You should ask youserlf when coding: what kind of error can happen here? Depending on that you can then either catch different exceptions in there, or then use Bad request - but its also not fix-all solution, one should reflect why they are doing a try catch here and what they are trying to catch.

One should also think about when to use the try catch - its not something to just everywhere either.

Reply:
If you dont know what can happen, you can just remove the try catch block and see what happen, we'll catch the possible error when / if such a thing can happen here.